include/seastar/core/future.hh

```diff
@@ -234,8 +234,9 @@ struct future_state_base {
     enum class state : uintptr_t {
          invalid = 0,
          future = 1,
-         result = 2,
-         exception_min = 3,  // or anything greater
+         result_unavailable = 2,
```

Comment:
The purpose of this extra state is solely to mimic the behaviour previosly imposed by setting `state::invalid` in `any(any&& x)` on `get_available_state()`.

Reply:
!result_unavailable` and `result` share the common bit intentionally.